backends/jwt_remote.go

```diff
@@ -229,9 +260,31 @@ func (o *remoteJWTChecker) jwtRequest(host, uri, token string, dataMap map[strin
 		tlsStr = "https://"
 	}
 
-	fullURI := fmt.Sprintf("%s%s%s", tlsStr, o.host, uri)
+	host = o.host
+	if host == "" && o.options.parseToken {
+		iss, err := getIssForToken(o.options, token, o.options.skipUserExpiration)
+		if err == nil {
+			if o.isHostWhitelisted(iss) {
+				host = iss
+			} else {
+				errorString := fmt.Sprintf("host %s obtained from jwt is not whitelisted; rejecting", iss)
+				log.Error(errorString)
+				return false, errors.New(errorString)
+			}
+		} else {
+			log.Printf("jwt remote get iss error: %s", err)
+		}
+	}
+
+	if host == "" {
+		errorString := fmt.Sprintf("cannot determine host for the authorization request for token %s", token)
+		log.Error(errorString)
+		return false, errors.New(errorString)
+	}
+
+	fullURI := fmt.Sprintf("%s%s%s", tlsStr, host, uri)
 	if o.port != "" {
-		fullURI = fmt.Sprintf("%s%s:%s%s", tlsStr, o.host, o.port, uri)
+		fullURI = fmt.Sprintf("%s%s:%s%s", tlsStr, host, o.port, uri)
```

Comment:
What if the user passed a hosts whitelists, we matched one and it included a port, but mistakenly also set the `port` option?
We could have `getIssForToken` check for the presence of a port to set it in that case and avoid this problem. Thoughts?

Reply:
This could come in handy to extract the hostname and the port parts: https://pkg.go.dev/net/url#URL

Reply:
I've just added simple check for ":" in the "host" variable - it seems to be enough and more logical, since we are checking what we have in this variable in the same place where we are constructing URI for the request. But if you thnik that this is not a good enough solution, I can add something more sophisticated :)

Reply:
That seems fine for now, thanks.